content/courses/intro-to-ada/chapters/tasking.rst

```diff
@@ -47,14 +47,17 @@ that both calls to :ada:`Put_Line` are performed.
 
 Note that:
 
-- The main application is itself a task (the main task).
+- The main application is itself a task (the main or “environment” task).
 
   - In this example, the subprogram :ada:`Show_Simple_Task` is the main task of
     the application.
 
 - Task :ada:`T` is a subtask.
 
-  - Each subtask has a master task.
+  - Each subtask has a master, which represents the program construct in which
```

Comment:
Two comments here: 

1. There are four "which"s in this paragraph. Maybe you could rewrite this?
2. The term "construct" (as in "program construct" or "master construct") won't be understood by everyone. As it appears here for the first time, you probably should explain it in a sentence.

Reply:
The term "construct" is very general.  This is probably not the place to introduce it.  I would suggest you introduce it early on in your Introduction to Ada. It is defined in 1.1.4 of the (Annotated) Ada Reference Manual:

> 15
> A syntactic category is a nonterminal in the grammar defined in BNF under “Syntax”. ...
> 16
> A construct is a piece of text (explicit or implicit) that is an instance of a syntactic category defined under “Syntax”. 

> ...
> 16.b
> Ramification: For example, an [expression](http://www.ada-auth.org/standards/2xaarm/html/AA-4-4.html#S0132) is a construct. A declaration is a construct, whereas the thing declared by a declaration is an “entity”. 
> ...

> 17
> A constituent of a construct is the construct itself, or any construct appearing within it.

Reply:
And feel free to break the long sentence into two, to avoid the multiple "which"es.

Reply:
The term "construct" is already being used in the course (and more often than I expected), so it won't change much if we use it here.